apps/dashboard/src/components/editor/use-editor-content.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from '../../data';
+import UnpublishedChangesNotice from './unpublished-changes-notice';
+import { EDITOR_VIEW_DRAFT } from '../../util/editor';
+
+// Do something about the timeout - if no project ID should be 500 ?
+export const useEditorContent = ( projectId, editorView ) => {
+	const [ ready, setReady ] = useState( ! projectId );
+
+	const { removeNotice } = useDispatch( 'core/notices' );
+	const {
+		updateEditorContent,
+		restoreEditorContent,
+		setEditorContentChanged,
+	} = useDispatch( STORE_NAME );
+
+	// If projectId or editorView changes, reset 'ready'
+	useEffect( () => {
+		setReady( false );
+	}, [ projectId, editorView ] );
+
+	return useCallback(
+		// using debounce here so we don't pound the state synchronization
+		debounce( ( content ) => {
+			// Isolated block editor forces a save as soon as the editor content has loaded.
+			// Ignore the first save and set 'ready' to true.
+			//
+			// https://github.com/Automattic/isolated-block-editor/blob/bca504ae1ef98cf1aba136d70e29fc339aa8ec61/src/components/content-saver/index.js#L47
+			if ( ! ready ) {
+				// editorView changes via setForceDraft to EDITOR_VIEW_DRAFT, meaning a restore intent
+				if ( editorView === EDITOR_VIEW_DRAFT ) {
+					restoreEditorContent( content );
+				}
+				setReady( true );
+				return;
+			}
+
+			// Mark editor content as changed
+			setEditorContentChanged();
```

Comment:
I guess now that we have actions for saving the actual changes, we don't need `setEditorContentChanged()` anymore - the reducer can just update the flag when any of the other actions is dispatched.